### PR TITLE
Wait for at least one account to have been created before starting the account benchmarks

### PR DESCRIPTION
**Pull request by steveluscher**
#### Problem

Consider a benchmark like the one in #3959. Presently, that PR causes a lot of this to be echoed when the benchmark first starts up:

```shell
[2024-12-05T22:22:43.421033375Z INFO  solana_accounts_cluster_bench] get_account_info: No accounts have yet been created; skipping
[2024-12-05T22:22:43.421044166Z INFO  solana_accounts_cluster_bench] get_account_info: No accounts have yet been created; skipping
[2024-12-05T22:22:43.421054506Z INFO  solana_accounts_cluster_bench] get_account_info: No accounts have yet been created; skipping
[2024-12-05T22:22:43.421065336Z INFO  solana_accounts_cluster_bench] get_account_info: No accounts have yet been created; skipping
[2024-12-05T22:22:43.421075706Z INFO  solana_accounts_cluster_bench] get_account_info: No accounts have yet been created; skipping
```

#### Summary of Changes

Wait on the creation of the first accounts before starting the benchmark.

Related: #3959.

### CODE REVIEW

File: accounts-cluster-bench/src/main.rs
Comment (steveluscher) [inline accounts-cluster-bench/src/main.rs]: Prevents `run_rpc_bench_loop` from being executed until it receives the signal.

File: accounts-cluster-bench/src/main.rs
Comment (steveluscher) [inline accounts-cluster-bench/src/main.rs]: Sits in the account creation loop, just after the creation block.

Comment (alessandrod) [reply accounts-cluster-bench/src/main.rs]: This will only unblock one worker right? You need a Barrier or you can use a channel and drop the sender to make recv fail (and handle the failure by just continuing)

**Comment by steveluscher** [comment]
Updated implementation to use `Barrier` for synchronization.

**Review by t-nelson — APPROVED** [review]

**Comment by behzadnouri** [comment]
@steveluscher Can you please do not merge with red CI.
This has broken CI for others:
https://buildkite.com/anza/agave/builds/15530#0193ac9c-ff2e-4c03-a763-be27344672ab

@yihau didn't we have a check to prevent people to merge without a green CI.

**Comment by steveluscher** [comment]
> Can you please do not merge with red CI.

Wild. I'm really diligent about this. Let me see what happened here.

**Comment by steveluscher** [comment]
I see what happened. I had a ton of PRs in a stack, none of which pushed this method over the limit itself, but the _rebased_ version of this one (on top of one of the others) did.

I'll send a PR to fix that now.

**Comment by steveluscher** [comment]
#4023.

**Pull request merged**
